homeassistant/components/twinkly/light.py

```diff
@@ -127,19 +132,41 @@ def device_info(self) -> DeviceInfo | None:
             name=self.name,
         )
 
+    @property
+    def supported_features(self) -> LightEntityFeature:
+        """Return supported features."""
+        return LightEntityFeature.EFFECT
+
     @property
     def is_on(self) -> bool:
         """Return true if light is on."""
         return self._is_on
 
     @property
-    def extra_state_attributes(self) -> dict:
+    def extra_state_attributes(self) -> Mapping[str, Any]:
         """Return device specific state attributes."""
 
         attributes = self._attributes
 
         return attributes
 
+    @property
+    def effect(self) -> str | None:
+        """Return the current effect."""
+        if "name" in self._current_movie:
+            _LOGGER.debug("Current effect '%s'", self._current_movie["name"])
```

Comment:
Please don't do side effects, like logging, in entity state properties. Either move the logging to a callback or remove it.

Reply:
https://github.com/home-assistant/core/pull/82915